metrics/ci_integrations/test/source/github_actions/config/validation_delegate/github_actions_source_validation_delegate_test.dart

```diff
@@ -0,0 +1,424 @@
+    const id = 1;
+    const repositoryOwner = 'owner';
+    const repositoryName = 'name';
+    const workflowId = 'workflow_id';
+
+    const githubToken = GithubToken(
+      scopes: [
+        GithubTokenScope.repo,
+      ],
+    );
+
+    const githubUser = GithubUser(
+      id: id,
+      login: repositoryOwner,
+    );
+
+    const githubRepository = GithubRepository(
+      id: id,
+      name: repositoryName,
+      owner: githubUser,
+    );
+
+    const githubActionsWorkflow = GithubActionsWorkflow(
+      path: workflowId,
+    );
+
+    final auth = BearerAuthorization('token');
+
+    final client = GithubActionsClientMock();
+    final delegate = GithubActionsSourceValidationDelegate(client);
+
+    tearDown(() {
+      reset(client);
+    });
+
+    test(
+      "throws an ArgumentError if the given client is null",
+      () {
+        expect(
+          () => GithubActionsSourceValidationDelegate(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given client",
+      () {
+        expect(
+          () => GithubActionsSourceValidationDelegate(client),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".validateAuth() returns an error if the interaction with the client is not successful",
+      () async {
+        when(
+          client.fetchToken(auth),
+        ).thenErrorWith();
+
+        final interactionResult = await delegate.validateAuth(auth);
+
+        expect(interactionResult.isError, isTrue);
+      },
+    );
+
+    test(
+      ".validateAuth() returns an interaction with the token invalid message if the interaction with the client is not successful",
+      () async {
+        when(
+          client.fetchToken(auth),
+        ).thenErrorWith();
+
+        final interactionResult = await delegate.validateAuth(auth);
+        final message = interactionResult.message;
+
+        expect(message, equals(GithubActionsStrings.tokenInvalid));
+      },
+    );
+
+    test(
+      ".validateAuth() returns an error if the result of an interaction with the client is null",
+      () async {
+        when(
+          client.fetchToken(auth),
+        ).thenSuccessWith(null);
+
+        final interactionResult = await delegate.validateAuth(auth);
+
+        expect(interactionResult.isError, isTrue);
+      },
+    );
+
+    test(
+      ".validateAuth() returns an interaction with the token invalid message if the result of an interaction with the client is null",
+      () async {
+        when(
+          client.fetchToken(auth),
+        ).thenSuccessWith(null);
+
+        final interactionResult = await delegate.validateAuth(auth);
+        final message = interactionResult.message;
+
+        expect(message, equals(GithubActionsStrings.tokenInvalid));
+      },
+    );
+
+    test(
+      ".validateAuth() returns an error if the fetched token does not have the required scope",
```

Comment:
Should we test that this method returns an error when the fetches token scopes are `null`?